compiler/optimizer/OMROptimizationManager.cpp

```diff
@@ -49,7 +49,7 @@ OMR::OptimizationManager::OptimizationManager(TR::Optimizer *o, OptimizationFact
         _numPassesCompleted(0),
         _optData(NULL),
         _optPolicy(NULL),
-        _enabled(false),
+        _enabled(!o->comp()->isDisabled(optNum)),
```

Comment:
I am not sure that is enough... the optimizer dispatch loop checks a regex as well as the isDisabled call (see line 1624 onward in OMROptimizer.cpp).

If we are going to standardize this logic then optimizer invocation sequence should be using this same query for its early exit at the point indicated.

Further rather than always writing optimizer()->isEnabled.... wouldn't it make sense for the root Optimization to gain an enabled() or disabled() function that wraps up this logic so it is easy to redirect the handling again if desired in the future?

A compile-time verification is needed to make sure that the refactoring isn't going to increase compile time.